testutils,kvserver: disable store rebalancer when replication manual

kvoli opened the pull request:
`ReplicationManual` can be specified in tests which use a test cluster to prevent lease/replica movements.

This proves vital if the test is asserting exactly on the placement of a range, or its leaseholder.

Also disable the store rebalancer, which could move both leases and replicas under `ReplicationManual` before.

Informs: #132310
Informs: #132272
Release note: None

kvoli committed:
testutils,kvserver: disable store rebalancer when replication manual

`ReplicationManual` can be specified in tests which use a test cluster
to prevent lease/replica movements.

This proves vital if the test is asserting exactly on the placement of a
range, or its leaseholder.

Also disable the store rebalancer, which could move both leases and
replicas under `ReplicationManual` before.

Informs: #132310
Informs: #132272
Release note: None

pav-kv commented:
Does this eliminate all lease movements? If so, consider also removing the pinned lease that I added in [this](https://github.com/cockroachdb/cockroach/blob/ea1419cb12ff5e16ac50f72ee7d58bd00a001b89/pkg/kv/kvserver/flow_control_integration_test.go#L3486-L3488) RACv2 test recently.

kvoli commented:
> Does this eliminate all lease movements? If so, consider also removing the pinned lease that I added in [this](https://github.com/cockroachdb/cockroach/blob/ea1419cb12ff5e16ac50f72ee7d58bd00a001b89/pkg/kv/kvserver/flow_control_integration_test.go#L3486-L3488) RACv2 test recently.

Good idea, I updated https://github.com/cockroachdb/cockroach/pull/132571 to do this. I'll leave this PR as is.

andrewbaptist reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/132590#-:-O9BYA84EfW0ECkArq7S:bg7igkv)* status: <span>:shipit:</span> complete! 2 of 0 LGTMs obtained (waiting on @pav-kv)



<!-- Sent from Reviewable.io -->

kvoli commented:
TYFTRs!

bors r=andrewbaptist,sumeerbhola
